test/cppsuite/test_harness/connection.h

```diff
@@ -0,0 +1,109 @@
+/*-
+ * Public Domain 2014-present MongoDB, Inc.
+ * Public Domain 2008-2014 WiredTiger, Inc.
+ *
+ * This is free and unencumbered software released into the public domain.
+ *
+ * Anyone is free to copy, modify, publish, use, compile, sell, or
+ * distribute this software, either in source code form or as a compiled
+ * binary, for any purpose, commercial or non-commercial, and by any
+ * means.
+ *
+ * In jurisdictions that recognize copyright laws, the author or authors
+ * of this software dedicate any and all copyright interest in the
+ * software to the public domain. We make this dedication for the benefit
+ * of the public at large and to the detriment of our heirs and
+ * successors. We intend this dedication to be an overt act of
+ * relinquishment in perpetuity of all present and future rights to this
+ * software under copyright law.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#ifndef CONN_API_H
+#define CONN_API_H
+
+#include <mutex>
+
+extern "C" {
+#include "test_util.h"
+#include "wiredtiger.h"
+}
+
+#include "api_const.h"
+#include "debug_utils.h"
+
+namespace test_harness {
+/*
+ * Singleton class owning the database connection, provides access to sessions and any other
+ * required connection API calls.
+ */
+class connection {
+    public:
+    /* No copies of the singleton allowed. */
+    connection(connection const &) = delete;
+    void operator=(connection const &) = delete;
+
+    static connection &
+    instance()
+    {
+        static connection _instance;
+        return _instance;
+    }
+
+    void
+    close()
+    {
+        if (_conn != nullptr) {
+            testutil_check(_conn->close(_conn, NULL));
+            _conn = nullptr;
+        }
+    }
+
+    void
+    create(std::string home = DEFAULT_DIR)
+    {
+        if (_conn != nullptr) {
+            debug_info("Connection is not NULL, cannot be re-opened.", _trace_level, DEBUG_ERROR);
+            testutil_die(EINVAL, "Connection is not NULL");
+        }
+
+        /* Create the working dir. */
+        testutil_make_work_dir(home.c_str());
+
+        /* Open conn. */
+        testutil_check(wiredtiger_open(home.c_str(), NULL, CONNECTION_CREATE, &_conn));
+    }
+
+    WT_SESSION *
+    get_session()
```

Comment:
Should this be open_session instead?

Reply:
It is true that the title could be misleading, we thought about it with Luke. However the user does not always need to know what's happening behind the scenes. We should either add an explicit comment in the function header or change the title.

Reply:
Sure 👍